src/mon/MgrMonitor.cc

```diff
@@ -987,6 +987,29 @@ int MgrMonitor::dump_metadata(const string& name, Formatter *f, ostream *err)
   return 0;
 }
 
+void MgrMonitor::print_nodes(Formatter *f)
+{
+  assert(f);
+
+  std::map<string, list<int> > mgrs;    // hostname => mgr
+  std::set<string> ls = map.get_all_names();
+  for (auto& name : ls) {
+    std::map<string,string> meta;
+    if (load_metadata(name, meta, NULL)) {
+      continue;
+    }
+   std::map<string, string>::iterator hostname = meta.find("hostname");
+    if (hostname == meta.end()) {
+      // not likely though
+      continue;
+    }
+    // FIXME: not sure any mgrmap_id valid here?
+    mgrs[hostname->second].push_back(0);
```

Comment:
Seems no suitable id related to mgr could push_back here, need you Ceph experts hint.